app/controller/Groups.js

```diff
@@ -400,23 +400,21 @@ Ext.define('DevCycleMobile.controller.Groups', {
 		var riderRecord = riderStore.first();
 		var thisRiderId = riderRecord.get("riderId");
 
-
 		var canCreateGroup = false;
-		if(groupName != '' && groupName.length <= NAME_MAX) {
-			if(groupCode == '') {
-				// generate random code 
+		if(groupName !== '' && groupName.length <= NAME_MAX) {
+			if(groupCode === '') {
+				// generate random code
 				groupCode = Math.random().toString(36).slice(2).substring(0,3);
 				Ext.getCmp('create_group_code').setValue(groupCode.toUpperCase());
 				canCreateGroup = true;
-			}
-			else if(groupCode.length >= CODE_MIN && groupCode.length <= CODE_MAX) {
+			} else if((new RegExp("\\\w{"+CODE_MIN+","+CODE_MAX+"}$")).test(groupCode)) {
```

Comment:
me too :)
Thanks for the catch, been doing too much java lately :stuck_out_tongue_winking_eye: